src/gauging.jl

```diff
@@ -1,39 +1,66 @@
-"""initialize bond tensors of an ITN to identity matrices"""
-function initialize_bond_tensors(ψ::ITensorNetwork; index_map=prime)
-  bond_tensors = DataGraph{vertextype(ψ),Nothing,ITensor}(underlying_graph(ψ))
+function default_bond_tensor_cache(ψ::ITensorNetwork)
+  return DataGraph{vertextype(ψ),Nothing,ITensor}(underlying_graph(ψ))
+end
+default_bond_tensor(index) = denseblocks(delta(index, prime(index)))
+
+struct VidalITensorNetwork{V,BTS} <: AbstractITensorNetwork{V}
+  itensornetwork::ITensorNetwork{V}
+  bond_tensors::BTS
+end
 
+itensornetwork(ψv::VidalITensorNetwork) = ψv.itensornetwork
+bond_tensors(ψv::VidalITensorNetwork) = ψv.bond_tensors
+bond_tensor(ψv::VidalITensorNetwork, e) = bond_tensors(ψv)[e]
+
+data_graph_type(::Type{VidalITensorNetwork}) = data_graph_type(itensornetwork(ψv))
+data_graph(ψv::VidalITensorNetwork) = data_graph(itensornetwork(ψv))
+function copy(ψv::VidalITensorNetwork)
+  return VidalITensorNetwork(copy(itensornetwork(ψv)), copy(bond_tensors(ψv)))
+end
+
+function VidalITensorNetwork(
+  ψ::ITensorNetwork;
+  bond_tensors_cache=default_bond_tensor_cache,
+  bond_tensor=default_bond_tensor,
+)
```

Comment:
I find this function a bit confusing, which relates to my comment below about the convention behind whether a certain function updates/improves the gauge/cache vs. just performs a conversion between formats.

If I were to guess what this function was doing without looking at the code, I would guess that `VidalITensorNetwork(ψ::ITensorNetwork)` might run BP on `ψ`, and then convert to the Vidal gauge using the BP message tensors. Instead, this just inserts identity tensors as the bond tensors. Instead of choosing between those two, I think it would make sense for this function to optionally accept a BP cache, optionally update that BP cache, and then convert to the Vidal gauge.

So, if no BP cache is passed, and the user specifies they don't want BP to be run, then it could fall back to this current behavior (using identity tensors as bond tensors).